testing/mariadb-connector-odbc: new aport

jjsearle opened the pull request:
https://downloads.mariadb.org/connector-odbc/
MariaDB Connector/ODBC is a standardized, LGPL licensed database driver using the industry standard ODBC API.

jjsearle commented:
resolves the request I added here
https://gitlab.alpinelinux.org/alpine/aports/issues/10779

jjsearle commented:
This is my first package build, please let me know if I need to make any improvements.

maxice8 commented on testing/mariadb-connector-odbc/APKBUILD:
```diff
@@ -0,0 +1,37 @@
+license="LGPL-2.1-only"
+depends="unixodbc"
+makedepends="cmake unixodbc unixodbc-dev mariadb-connector-c mariadb-connector-c-dev mariadb-static"
+install=""
```
empty variables can be removed for brevity

jjsearle replied:
empty variables removed.

maxice8 commented on testing/mariadb-connector-odbc/APKBUILD:
```diff
@@ -0,0 +1,37 @@
+depends="unixodbc"
+makedepends="cmake unixodbc unixodbc-dev mariadb-connector-c mariadb-connector-c-dev mariadb-static"
+install=""
+source="https://downloads.mariadb.org/interstitial/connector-odbc-3.1.3/mariadb-connector-odbc-3.1.3-ga-src.tar.gz CMakeList.txt.patch"
```
have one source per-line.

Don't hardcode the version, use $pkgver

jjsearle replied:
> have one source per-line.
Done this

> Don't hardcode the version, use $pkgver
Done this

Cogitri commented on testing/mariadb-connector-odbc/APKBUILD:
```diff
@@ -0,0 +1,37 @@
+		${CMAKE_CROSSOPTS} .
+	make
+}
+
```
Are there tests? If so, please run them in `check()`, if not please add `options="!check"` to the APKBUILD to indicate that.

jjsearle replied:
There are tests in the git repository but they are not included in the tagged releases.

`options="!check" ` has been added.

jjsearle commented on testing/mariadb-connector-odbc/CMakeList.txt.patch:
```diff
@@ -0,0 +1,10 @@
+ 
+ INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/libmariadb/include)
+ INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR}/libmariadb/include)
++INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR} /usr/include/mysql)
```
No, as I thought this was just something necessary for alpine. Without this patch it get the following when building.
```
/home/joe/aports/testing/mariadb-connector-odbc/src/mariadb-connector-odbc-3.1.4/ma_odbc.h:32:10: fatal error: mysql.h: No such file or directory
 #include <mysql.h>
          ^~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/maodbc.dir/build.make:63: CMakeFiles/maodbc.dir/odbc_3_api.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /home/joe/aports/testing/mariadb-connector-odbc/src/mariadb-connector-odbc-3.1.4/ma_error.c:19:
/home/joe/aports/testing/mariadb-connector-odbc/src/mariadb-connector-odbc-3.1.4/ma_odbc.h:32:10: fatal error: mysql.h: No such file or directory
 #include <mysql.h>
          ^~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/maodbc.dir/build.make:76: CMakeFiles/maodbc.dir/ma_error.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/maodbc.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
>>> ERROR: mariadb-connector-odbc: build failed
```
There is possibly a better way of pointing cmake at the correct files but it's not something I have much experience with. I am open to suggestions.

Cogitri replied:
It'd be best to file an issue about this upstream then, I doubt it only applies to Alpine.

jjsearle replied:
I have raised this upstream here: https://jira.mariadb.org/browse/ODBC-265

Cogitri replied:
Thank you. Could you also add a note to the patch for future reference?

jjsearle replied:
done

Cogitri commented on testing/mariadb-connector-odbc/APKBUILD:
```diff
@@ -0,0 +1,38 @@
+arch="all"
+license="LGPL-2.1-only"
+depends="unixodbc"
+makedepends="cmake unixodbc unixodbc-dev mariadb-connector-c mariadb-connector-c-dev mariadb-static"
```
Are really both mariadb-connector-c and mariadb-connector-c-dev required?

jjsearle replied:
`mariadb-connector-c-dev` is required but that package depends upon `mariadb-connector-c`. I have removed `mariadb-connector-c` as a make dependency as `mariadb-connector-c-dev` will pull it down as a dependency anyway.

I have also removed `unixodbc` as a make dependency as that isn't required.

Cogitri commented:
I'd recommend you to run `apkbuild-lint` from the `atools` package on the `APKBUILD` too:

```
pre-commit: MC:[AL1]:testing/mariadb-connector-odbc/APKBUILD:14:builddir can be removed as it is the default value
pre-commit: MC:[AL29]:testing/mariadb-connector-odbc/APKBUILD:12:$pkgname should not be used in the source url
pre-commit: MP:[AL32]:testing/mariadb-connector-odbc/APKBUILD:12:unnecesary usage of braces: ${pkgname}
pre-commit: MP:[AL32]:testing/mariadb-connector-odbc/APKBUILD:12:unnecesary usage of braces: ${pkgname}
pre-commit: MP:[AL32]:testing/mariadb-connector-odbc/APKBUILD:12:unnecesary usage of braces: ${pkgver}
pre-commit: MP:[AL32]:testing/mariadb-connector-odbc/APKBUILD:12:unnecesary usage of braces: ${pkgver}
```

Maybe also add a comment the `options="!check"` to explain why they're disabled to help other people who might work on this APKBUILD. Otherwise this LGTM to me, thank you for your work

Cogitri commented:
Please use apkbuild-lint as advised in my earlier comment.

jjsearle commented:
I have : 
* corrected the issues flagged by the linter (nice tool :+1: ).
* added a note about why checks are disabled
* added a note to the patch with a lint to the upstream report

jjsearle commented:
I have removed the need for the patch by using
```
-DCMAKE_C_FLAGS_RELEASE="-I/usr/include/mysql" \
```
As recommended by upstream.
